iOS Client/DevotionsController/MBDevotionsViewController.swift

```diff
@@ -64,5 +75,20 @@ class MBDevotionsViewController: UIViewController, StoreSubscriber, UITableViewD
     func numberOfSections(in tableView: UITableView) -> Int {
         return 1
     }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let devotion = devotions[indexPath.row]
+        MBStore.sharedStore.dispatch(SelectedDevotion(devotion: devotion.devotion))
+        let store = MBStore()
+        store.markDevotionAsRead(date: devotion.devotion.date) { error in
+            if error == nil {
+                store.syncDevotions { devotions, error in
+                    if error == nil, let loadedDevotions = devotions {
+                        MBStore.sharedStore.dispatch(LoadedDevotions(devotions: .loaded(data: loadedDevotions)))
```

Comment:
sketchy to do this in a completion block b/c you don't know what thread you're on. This needs to be in a dispatch to the main thread (or, my vote is to make markDevotionAsRead just return an error type and then you can check the error and do this imperatively on the next line. No need for this completion block approach IMO since this is not async code)